src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketClientFactoryImpl.java

```diff
@@ -27,6 +29,18 @@ public BitbucketCapabilitiesClient getCapabilityClient() {
         return new BitbucketCapabilitiesClientImpl(bitbucketRequestExecutor);
     }
 
+    @VisibleForTesting
+    public BitbucketBuildStatusClient getBuildStatusClient(String revisionSha,
+                                                           @Nullable BitbucketSCM bitbucketSCM,
+                                                           BitbucketCICapabilities ciCapabilities,
+                                                           InstanceIdentityProvider instanceIdentityProvider) {
```

Comment:
We decided to keep `com.atlassian.bitbucket.jenkins.internal.client` free from Jenkins reference. This will help in keeping our client package modular. It will be super easy to port this to another CI like circle CI, Team city if we wish to. I would suggest introducing an interface which provides Key Pair. Implementation can be instance identity-based.